ccpp/CCPP_driver.F90

```diff
@@ -205,38 +237,61 @@ subroutine CCPP_step (step, nblks, ierr)
-          if (ierr2/=0) then
-            write(0,'(2a,3(a,i4),a)') "An error occurred in ccpp_physics_run for group ", "phys_ts", &
-                                      ", block/chunk ", nb, " and thread ", nt, " (ntX=", ntX, "):"
-            write(0,'(a)') trim(cdata_block(nb,ntX)%errmsg)
-            ierr = ierr + ierr2
-          endif
-        else
-          if (trim(step)=="radiation") then
+           if (trim(dycore)=="fv3") then
+              ! Reset GFS_Interstitial DDT physics fields for this thread
+              call GFS_Interstitial(ntX)%phys_reset(GFS_control)
+              ! Process-split physics
+              call ccpp_physics_run(cdata_block(nb,ntX), suite_name=trim(ccpp_suite), group_name="phys_ps", ierr=ierr2)
+              if (ierr2/=0) then
+                 write(0,'(2a,3(a,i4),a)') "An error occurred in ccpp_physics_run for group ", "phys_ps", &
+                                           ", block/chunk ", nb, " and thread ", nt, " (ntX=", ntX, "):"
+                 write(0,'(a)') trim(cdata_block(nb,ntX)%errmsg)
+                 ierr = ierr + ierr2
+              endif
+              ! Time-split physics
+              call ccpp_physics_run(cdata_block(nb,ntX), suite_name=trim(ccpp_suite), group_name="phys_ts", ierr=ierr2)
+              if (ierr2/=0) then
+                 write(0,'(2a,3(a,i4),a)') "An error occurred in ccpp_physics_run for group ", "phys_ts", &
+                                           ", block/chunk ", nb, " and thread ", nt, " (ntX=", ntX, "):"
+                 write(0,'(a)') trim(cdata_block(nb,ntX)%errmsg)
+                 ierr = ierr + ierr2
+              endif
+           endif
+           if (trim(dycore)=="mpas") then
+              ! Physics
+              call ccpp_physics_run(cdata_block(nb,ntX), suite_name=trim(ccpp_suite), group_name="physics", ierr=ierr2)
+              if (ierr2/=0) then
+                 write(0,'(2a,3(a,i4),a)') "An error occurred in ccpp_physics_run for group ", "physics", &
+                                           ", block/chunk ", nb, " and thread ", nt, " (ntX=", ntX, "):"
+                 write(0,'(a)') trim(cdata_block(nb,ntX)%errmsg)
+                 ierr = ierr + ierr2
+              endif
+           endif
+        endif
+        if (trim(step)=="microphysics") then
+           if (trim(dycore)=="mpas") then
+              ! Microphysics
+              call ccpp_physics_run(cdata_block(nb,ntX), suite_name=trim(ccpp_suite), group_name="microphysics", ierr=ierr2)
+              if (ierr2/=0) then
+                 write(0,'(2a,3(a,i4),a)') "An error occurred in ccpp_physics_run for group ", "microphysics", &
+                                           ", block/chunk ", nb, " and thread ", nt, " (ntX=", ntX, "):"
+                 write(0,'(a)') trim(cdata_block(nb,ntX)%errmsg)
+                 ierr = ierr + ierr2
+              endif
+           endif
```

Comment:
Should there be an `else` statement that raises an error, since `microphysics` is only valid for `mpas` at the moment?

Reply:
Good point. I will add an error.

Reply:
Done.